src/client/handlers/VaultsSecretsList.ts

```diff
@@ -40,18 +37,37 @@ class VaultsSecretsList extends ServerHandler<
       return await vaultManager.withVaults(
         [vaultId],
         async (vault) => {
-          return await vaultOps.listSecrets(vault);
+          const data = await vault.readF(async (fs) => {
+            const fileTreeGen = fileTree.globWalk({
+              fs: fs,
+              basePath: '.',
+              pattern: input.pattern,
+              yieldStats: true,
+              yieldRoot: false,
+              yieldFiles: true,
+              yieldParents: true,
+              yieldDirectories: true,
+            });
+            const data: Array<TreeNode | ContentNode | Uint8Array> = [];
+            const serializedStream = fileTree.serializerStreamFactory(
+              fs,
+              fileTreeGen,
+              true,
+            );
+            for await (const output of serializedStream) {
+              data.push(output);
+            }
+            return data;
+          });
```

Comment:
This is a bad pattern, everything until now has been designed to be streamed. So we don't want to accumulate the stream as a single array of strings. You turn something that streams parts in piece meal into one gigantic message here.

We want to avoid that so the call can scale. For example what happens if the tree is really big and contains 10gb of data? Do we want to wait the whole time it takes to read all of that? Then store it all in memory while we format the message and send it over in one giant chunk? It's not going to work.

The Handler needs to be a `RawHandler` and `serializedStream` needs to be returned as the output raw stream.

I'm guessing you did this because `withVaults` and `readF` returns a single value. You need to use the generator forms of `withVaultsG` and `readG` to do this properly. If you don't know how js `AsyncGenerator`s work then be sure to read up on them.